AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -51,158 +58,121 @@
 /// </summary>
 public class AwsKmsHierarchicalKeyring
 {
-    // THESE ARE PUBLIC RESOURCES DO NOT USE IN A PRODUCTION ENVIRONMENT
-    private static string branchKeyIdA = "43574aa0-de30-424e-bad4-0b06f6e89478";
-    private static string branchKeyIdB = "a2f4be37-15ec-489a-bcb5-dcce1f6bfe84";
-    private static void Run(MemoryStream plaintext)
+    private void Run(MemoryStream plaintext)
     {
-        var kmsConfig = new KMSConfiguration { KmsKeyArn = ExampleUtils.ExampleUtils.GetBranchKeyArn() };
-        // Create an AWS KMS Configuration to use with your KeyStore.
-        // The KMS Configuration MUST have the right access to the resources in the KeyStore.
-        var keystoreConfig = new KeyStoreConfig
+        // 1. Configure your KeyStore resource.
+        //    `ddbTableName` is the name you want for the DDB table that
+        //    will back your keystore.
+        //    `kmsKeyArn` is the KMS Key that will protect your branch keys and beacon keys
+        //    when they are stored in your DDB table.
+        var keyStoreConfig = new KeyStoreConfig
         {
-            // Client MUST have permissions to decrypt kmsConfig.KmsKeyArn
             KmsClient = new AmazonKeyManagementServiceClient(),
-            KmsConfiguration = kmsConfig,
+            KmsConfiguration = new KMSConfiguration{ KmsKeyArn = ExampleUtils.ExampleUtils.GetBranchKeyArn() },
             DdbTableName = ExampleUtils.ExampleUtils.GetKeyStoreName(),
             DdbClient = new AmazonDynamoDBClient(),
             LogicalKeyStoreName = ExampleUtils.ExampleUtils.GetLogicalKeyStoreName() 
         };
-        var materialProviders = new MaterialProviders(new MaterialProvidersConfig());
-        var keystore = new KeyStore(keystoreConfig);
+        var keyStore = new KeyStore(keyStoreConfig);
+        
+        // 2. Create the DynamoDb table that will store the branch keys and beacon keys.
+        //    This checks if the correct table already exists at `ddbTableName`
+        //    by using the DescribeTable API. If no table exists,
+        //    it will create one. If a table exists, it will verify
+        //    the table's configuration and will error if the configuration is incorrect.
+        //    It may take a couple minutes for the table to become ACTIVE,
+        //    at which point it is ready to store branch and beacon keys.
+        
+        // keyStore.CreateKeyStore(new CreateKeyStoreInput());
```

Comment:
Question: Why should we not run this?

The Description of in 2 states that IF the table already exists, 
it will validate the table.

Therefore, it SHOULD BE safe to run this,
as it will NOT create a table.

Though we may need to grant the CI Role Describe Table permission.